src/sinol_make/commands/run/__init__.py

```diff
@@ -0,0 +1,513 @@
+# Modified version of https://sinol3.dasie.mimuw.edu.pl/oij/jury/package/-/blob/master/runner.py
+# Author of the original code: Bartosz Kostka <kostka@oij.edu.pl>
+# Version 0.6 (2021-08-29)
+
+from sinol_make.interfaces.BaseCommand import BaseCommand
+from sinol_make.interfaces.Errors import CompilationError
+from sinol_make.helpers import compile
+import sinol_make.util as util
+import yaml, os, collections, sys, re, math
+import multiprocessing as mp
+
+class Command(BaseCommand):
+	"""
+	Class for running current task
+	"""
+
+
+	def get_name(self):
+		return 'run'
+
+
+	def configure_subparser(self, subparser):
+		parser = subparser.add_parser(
+			'run', 
+			help='Run current task',
+			description='Run current task'
+		)
+		parser.add_argument('--programs', type=str, nargs='+',
+							help='programs to be run, for example prog/abc{b,s}*.{cpp,py}')
+		parser.add_argument('--tests', type=str, nargs='+',
+							help='tests to be run, for example in/abc{0,1}*')
+		parser.add_argument('--cpus', type=int,
+							help='number of cpus to use, you have %d avaliable' % mp.cpu_count())
+		parser.add_argument('--tl', type=float, help='time limit (in s)')
+		parser.add_argument('--ml', type=float, help='memory limit (in MB)')
+		parser.add_argument('--verbose', dest='verbose', action='store_true',
+							help='show detailed raport (status for every test in a group)')
+		parser.add_argument('--show_memory', dest='show_memory', action='store_true',
+							help='show memory usage in report')
+		parser.add_argument('--validate_subtasks',
+							dest='validate_subtasks', action='store_true',
+							help='validate subtasks defined in config.yml')
+		parser.add_argument('--validate_programs',
+							dest='validate_programs', action='store_true',
+							help='validate programs in prog/ directory')
+		parser.add_argument('--subtask_report', type=str,
+							help='file to store report from subtask validation (in markdown)')
+		parser.add_argument('--program_report', type=str,
+							help='file to store report from program executions (in markdown)')
+		parser.add_argument('--oiejq_path', type=str,
+		      				help='path to oiejq executable (default: `~/.local/bin/oiejq`)')
+		parser.add_argument('--c_compiler', type=str,
+		    				help='C compiler to use (default for Linux and Windows: gcc, default for Mac: gcc-{9-12})')
+		parser.add_argument('--cpp_compiler', type=str,
+		    				help='C++ compiler to use (default for Linux and Windows: g++, default for Mac: gcc-{9-12})')
+		parser.add_argument('--python_interpreter', type=str,
+		    				help='Python interpreter to use (default: python3)')
+		parser.add_argument('--java_compiler', type=str,
+		    				help='Java compiler to use (default: javac)')
```

Comment:
Maybe there is a way to set a default value to those flags, that is, when the user doesn't provide any value, it assumes the default one.

Reply:
There is, but what should be the default value for c++ compiler, since it can be g++ on linux or g++-{9-12} on MacOS?

Reply:
I don't know how the Python CLI library works, but in some languages/libraries it's possible to send a function instead of a value, which evaluates and returns a default value.

Reply:
And if there is no c++ compiler found should we just return g++?

Reply:
As we assume that this case shouldn't happen, an error with a suitable message might be better (explaining what the tool tried to do and that it's possible to manually set the command).

Reply:
f2b03c5f50acc72591fdf840326c3a669f17eec1